File: src/lib/types/db/library/LibraryArtist.ts
export interface LibraryArtist {
  id: string;
  library_id: number;
  artist_id: string;
  
  total_albums: number;
  total_tracks: number;
  total_duration: number;
  
  custom_image_url: string | null;
  notes: string | null;
  
  added_at: string;
}

File: src-tauri/src/entity/library/library_artist.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct LibraryArtist {
    pub id: String,
    pub library_id: i64,
    pub artist_id: String,
    
    pub total_albums: i64,
    pub total_tracks: i64,
    pub total_duration: f64,
    
    pub custom_image_url: Option<String>,
    pub notes: Option<String>,
    
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct LibraryArtistCreate {
    pub library_id: i64,
    pub artist_id: String,
}
